Makefile.am

```diff
@@ -0,0 +1,304 @@
+
+install-data-local:
+	$(MKDIR_P) $(DESTDIR)$(datarootdir)/applications
+	$(MKDIR_P) $(DESTDIR)$(datarootdir)/armory/img
+	rsync -rupE --exclude="img/.DS_Store" img $(DESTDIR)$(datarootdir)/armory/
+	for i in $(LANGUAGES); do $(MKDIR_P) $(DESTDIR)$(datarootdir)/locale/$$i/LC_MESSAGES; \
+	   cp$$i.mo $(DESTDIR)$(datarootdir)/locale/$$i/LC_MESSAGES/armory.mo; done
+
+install-exec-local:
+	$(MKDIR_P) $(DESTDIR)$(libdir)/armory/extras
+	$(MKDIR_P) $(DESTDIR)$(libdir)/armory/bitcoinrpc_jsonrpc
+	$(MKDIR_P) $(DESTDIR)$(libdir)/armory/txjsonrpc/web
+	$(MKDIR_P) $(DESTDIR)$(libdir)/armory/ui
+	$(MKDIR_P) $(DESTDIR)$(libdir)/armory/pytest
+	$(MKDIR_P) $(DESTDIR)$(libdir)/armory/BitTornado/BT1
+	$(MKDIR_P) $(DESTDIR)$(libdir)/armory/urllib3/contrib
+	$(MKDIR_P) $(DESTDIR)$(libdir)/armory/urllib3/packages/ssl_match_hostname
+	$(MKDIR_P) $(DESTDIR)$(libdir)/armory/armoryengine
+	$(MKDIR_P) $(DESTDIR)$(bindir)
+	$(SED) "s: /usr/lib: $(libdir):g" < dpkgfiles/armory > $(DESTDIR)$(bindir)/armory
+	chmod +x $(DESTDIR)$(bindir)/armory
+	cp $(ROOT_PY) $(ROOT_SO) README.md $(DESTDIR)$(libdir)/armory/
+	cp $(ARMORYENGINE_PY) $(DESTDIR)$(libdir)/armory/armoryengine
+	cp $(EXTRAS_PY) $(DESTDIR)$(libdir)/armory/extras
+	cp $(BITCOINRPC_JSONRPC_PY) $(DESTDIR)$(libdir)/armory/bitcoinrpc_jsonrpc
+	cp $(TXJSONRPC_PY) $(DESTDIR)$(libdir)/armory/txjsonrpc
+	cp $(TXJSONRPC_WEB_PY) $(DESTDIR)$(libdir)/armory/txjsonrpc/web
+	cp $(UI_PY) $(DESTDIR)$(libdir)/armory/ui
+	cp $(PYTEST_PY) $(DESTDIR)$(libdir)/armory/pytest
+	cp $(URLLIB3_PY) $(DESTDIR)$(libdir)/armory/urllib3
+	cp $(URLLIB3_CONTRIB_PY) $(DESTDIR)$(libdir)/armory/urllib3/contrib
+	cp $(URLLIB3_PACKAGES_PY) $(DESTDIR)$(libdir)/armory/urllib3/packages
+	cp $(URLLIB3_PACKAGES_SSL_MATCH_HOSTNAME_PY) \
+	   $(DESTDIR)$(libdir)/armory/urllib3/packages/ssl_match_hostname
+	cp $(BITTORNADO_PY) $(DESTDIR)$(libdir)/armory/BitTornado
+	cp $(BITTORNADO_BT1_PY) $(DESTDIR)$(libdir)/armory/BitTornado/BT1
+	cp default_bootstrap.torrent $(DESTDIR)$(libdir)/armory
+
+dpkgfiles/%.desktop: dpkgfiles/%.desktop.in Makefile
+	sed -e 's,[@]bindir[@],$(bindir),g' < $< > $@
+	chmod +x $@
+
+_CppBlockUtils.so: $(OBJS) cppForSwig/CppBlockUtils_wrap.o
+	$(LINK) -shared -fPIC $(LDFLAGS) $(CXXFLAGS) $(OBJS) cppForSwig/CppBlockUtils_wrap.o $(LDLIBS) -o $@
+
+$(OBJS):
+	$(MAKE) -C cppForSwig $(notdir $@)
+
+cppForSwig/CppBlockUtils_wrap.o:
+	$(MAKE) -C cppForSwig $(notdir $@)
+
+qrc_img_resources.py: imgList.xml
+	$(PYRCC4) -o $@ $<
+
+all-test-tools: all
+	$(MAKE) -C cppForSwig/gtest
+
+gtest: all-test-tools
+	(cd cppForSwig/gtest && ./CppBlockUtilsTests)
+
+pytest: all
+	python -m unittest discover
+
+test: gtest pytest
+
+po: $(patsubst %,po/%.po,$(LANGUAGES))
+mo: $(patsubst %,po/%.mo,$(LANGUAGES))
+
+po/%.po: po/messages.pot
+	msgmerge -U $@ $<
+
+po/%.mo: po/%.po po/messages.pot
+	msgfmt $< -o $@
+
+po/messages.pot: all $(MESSAGES_PY)
+	xgettext -o $@ --from-code=UTF-8 -L Python -ktr $(MESSAGES_PY)
+
+if TARGET_LINUX
+if BUILD_LINUX
+deb: all
```

Comment:
This needs to be slightly reworked.

- I'd just give the user one chance to supply a chroot name. If it's blank, skip it. If both get skipped, no worries.
- The integers need to be replaced with flags.